src/loaders/ImageLoader.js

```diff
@@ -50,7 +50,7 @@ ImageLoader.prototype = Object.assign( Object.create( Loader.prototype ), {
 
 			Cache.add( url, this );
 
-			if ( onLoad ) onLoad( this );
+			if ( onLoad ) onLoad( this || image );
```

Comment:
When might `this` be falsy? Won't it be the window or global object, if nothing else?

Reply:
It will be undefined if run this function without event, I guess.

`if ( url && url.substr( 0, 23) === 'data:' ) onImageLoad();`

Reply:
I don't think that's right, you can try this in node.js:

```js
fn = function () { return this; }
fn() // returns Global object
```

Reply:
I have another example. I think this is what is happening here.
```
function fnB() {
  return this; // returns undefined
}

function fnA() {
  fnB();
}

console.log(fnA());
```

Reply:
This is undefined here if call this function directly. I'm not sure why.
```
Cache.add( url, this );
console.log(this) // logs undefined
if ( onLoad ) onLoad( this || image );
```

I would suggest always return `image` instead of `this` here.
```
Cache.add( url, image );
if ( onLoad ) onLoad( image );
```